# Task#

I will be completing a task on creating and manipulating arrays. As well as analysing the data stored within the arrays. I will also be using numpy, which contains many mathematical functions that can be used to analyse the data stored in arrays. Examples of functions include sum, mean, sine, etc. I will also be using array slicing to reference a smaller subset of the original array, this can be useful for analysing very specific parts of an array.

# Part 1 #

In [1]:
import numpy as np #This imports the functions of numpy and it is shortened to np for ease of use

In [2]:
print(np.array([0,5,10]))

[ 0  5 10]


In [3]:
print(np.arange(0,11,5)) #Creates an array starting at 0, ending at 10 with a gap of 5

[ 0  5 10]


In [4]:
print(np.linspace(2.5,3.5,3)) #Creates an array starting at 2.5, ending at 3.5 with 3 evenly spaced values

[ 2.5  3.   3.5]


In [5]:
array_a = np.arange(0,11,5)
array_b = np.linspace(2.5,3.5,3)
array_c = array_a * array_b   #Multiplies the previously created arrays that are assigned to "array_a" and "array_b"
print(array_c)

[  0.  15.  35.]


# Difference Between Cross/Dot Product and * Operator #

In [6]:
print(np.dot(array_a,array_b)) #Calculates the dot product of "array_a" and "array_b"

50.0


In [7]:
print(np.cross(array_a,array_b)) #Calculates the cross product of "array_a" and "array_b"

[-12.5  25.  -12.5]


# Explanation of 'For' Loop #

In [8]:
r = np.linspace(1, 2.5, 5) #This creates an array starting at 1 and finishing at 2.5, with 5 values between 1 and 2.5 
print(r)                   #This prints the array that was created
for n in r:                #This picks, in order, each element in the array
    print("n is ", n)
    print("n*3 is ", n*3)  #For loop repeated after completing this line of code, as long as there are remaining elements
print("Finished")          #Last line of code run after For loop has finished with last element

[ 1.     1.375  1.75   2.125  2.5  ]
n is  1.0
n*3 is  3.0
n is  1.375
n*3 is  4.125
n is  1.75
n*3 is  5.25
n is  2.125
n*3 is  6.375
n is  2.5
n*3 is  7.5
Finished


The first line, as explained by a comment, creates the array starting at 1 and ending at 2.5. The array is populated with 5 values, which are equally spaced between 1 and 2.5. The next line prints the array, which can be seen above. The line after that picks each element in order from the array that was created and then the value of the element is printed in the form "n is x" where x is the value of the element. The next line of code prints the value of the element multiplied by 3. Once these 2 tasks are completed for an element, the process repeats itself for the next element, until there are no more elements left. Once there are no more elements left, "Finished" is printed as shown above.

# Part 2 #

In [9]:
array_d = np.loadtxt("session3data.txt") #This loads a text file containing values as an array
print(array_d)                           #This prints the imported array

[ 18.02  18.55  18.94  18.99  19.5   19.71  19.61  19.4   19.41  19.26
  18.88  18.71  18.42  17.84  17.01  16.86  16.27  15.58  15.25  15.04
  25.68  24.62  23.4   21.96  20.65]


# Manipulating An Array #

In [10]:
print(np.sin(array_d))

[-0.7376316  -0.29509593  0.09032082  0.13998283  0.60553987  0.75813222
  0.68924326  0.52306577  0.53156239  0.39901656  0.03043938 -0.13910337
 -0.41646711 -0.84659557 -0.96410101 -0.91359403 -0.53291073  0.12761433
  0.44212217  0.61938827  0.5203483  -0.4905678  -0.98691556  0.03114354
  0.97374664]


The function used above finds the sine of each element within the array, treating each of the values of the array in radians. The new array is printed.

In [11]:
print(np.size(array_d))

25


This function works out the size of the array in question. It calculates the number of elements along a given axis in the chosen array. As shown above, there are 25 elements in "array_d".

In [12]:
print(len(array_d))

25


The function used above gives the number of items in the specified array.

In [13]:
print(np.max(array_d))

25.68


This function prints the element of the array with the maximum value, in this case it is 25.68.

In [14]:
print(np.min(array_d))

15.04


The function used above prints the element of the array with the minimum value, in this case it is 15.04.

In [15]:
print(np.sum(array_d))

477.56


This function calculates the sum of all of the elements in the array, which was found to be 477.56.

In [16]:
print(np.mean(array_d))

19.1024


This function calculates the mean value of all the elements in the array, by calculating the sum of all the elements and then dividing by how many elements there are.

In [17]:
x = np.sum(array_d)
y = len(array_d)
print(x/y)

19.1024


The way I calculated the mean was by first finding the sum of all the elements in the array and assigning this number to "x", and then finding the number of elements in the array and assigning this number to "y". Then I printed the value for "x"/"y", which is the sum of all the elements divided by the number of elements, this gives the same mean as calculated previously.

# Array Slicing #

In [18]:
print(np.sum(array_d[:10]))

191.39


Using array slicing, I was able to reference only the first 10 elements of the array. The numbering in the array starts at 0, not 1, therefore the referenced subset of the array will include everything up to but not including #10. Then I enclosed the code for the sliced array within a sum function to find the sum of the first 10 elements. This value is then printed.

In [19]:
a = array_d[20:]
b = np.sum(a)
c = len(a)
print(b/c)

23.262


The sliced array includes all elements from and including #20 to the end of the array since there are 25 elements in total. This is assigned to "a". Then the sum of all the elements in "a" (the slice of the original array) is calculated and assigned to "b". Then the size of the slice of the array is calculated and assigned to "c". Then the mean is calculated by dividing the sum of the last 5 elements ("b") by the size of the 5 elements ("c"). This value is printed.